org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/AxesMetadataExample.java

```diff
@@ -0,0 +1,338 @@
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.DatasetUtils;
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.ILazyDataset;
+import org.eclipse.january.dataset.IndexIterator;
+import org.eclipse.january.dataset.LongDataset;
+import org.eclipse.january.dataset.Maths;
+import org.eclipse.january.metadata.AxesMetadata;
+import org.eclipse.january.metadata.internal.AxesMetadataImpl;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * example to demonstrate using axes metadata to combine non-synced measurements
+ * 
+ *
+ */
+public class AxesMetadataExample
+{
+
+  /**
+   * we wish the interpolation function to be ignorant of the specific mathematical operation being
+   * conducted. This interface wraps the operation in a Command
+   *
+   */
+  public static interface OperationPerformer
+  {
+    /**
+     * perform some operation on datasets and and b. If an output dataset is provided, store the
+     * results in there.
+     * 
+     * @param a
+     *          left-hand operand
+     * @param b
+     *          right-hand operand
+     * @param o
+     *          (optional) results object
+     * @return a dataset product
+     */
+    public Dataset perform(final Object a, final Object b, final Dataset o);
+  }
+
+  /**
+   * experiment with extending maths processing to be AxesMetadata aware
+   * 
+   * @author ian
+   *
+   */
+  private static class NewMaths extends Maths
+  {
+    /**
+     * 
+     * @param a
+     *          left-hand operand
+     * @param b
+     *          right-hand operand
+     * @param o
+     *          (optional) results object
+     * @param operation
+     *          operation to apply
+     * @return result of operation, resampling data if necessary to synchronise AxesMetadata
+     */
+    public static Dataset performWithInterpolation(final Object a,
+        final Object b, final Dataset o, final OperationPerformer operation)
+    {
+      final Object operandA;
+      final Object operandB;
+
+      // ok, check if we've received datasets
+      if (a instanceof Dataset && b instanceof Dataset)
+      {
+        final Dataset da = (Dataset) a;
+        final Dataset db = (Dataset) b;
+        final AxesMetadata axesA = da.getFirstMetadata(AxesMetadata.class);
+        final AxesMetadata axesB = db.getFirstMetadata(AxesMetadata.class);
+
+        if (axesA != null && axesB != null)
+        {
+          // ok, we've got indexed data. see if they match
+          final ILazyDataset aLazyIndices = axesA.getAxis(0)[0];
+          final ILazyDataset bLazyIndices = axesB.getAxis(0)[0];
+
+          // ok, if they're different length then we know they're non-synced
+          final boolean needInterp;
+          if (aLazyIndices.getSize() != bLazyIndices.getSize())
+          {
+            // ok, need syncing
+            needInterp = true;
+          }
+          else if (aLazyIndices.equals(bLazyIndices))
```

Comment:
If these are truly lazy then this check is flawed. Use the code from lines 118-9 first.